src/master-playlist-controller.js

```diff
@@ -545,27 +586,29 @@ export class MasterPlaylistController extends videojs.EventTarget {
    * @private
    */
   setupSegmentLoaderListeners_() {
-    this.mainSegmentLoader_.on('bandwidthupdate', () => {
-      const nextPlaylist = this.selectPlaylist();
-
-      if (this.shouldSwitchToMedia_(nextPlaylist)) {
-        this.masterPlaylistLoader_.media(nextPlaylist);
-      }
-
-      this.tech_.trigger('bandwidthupdate');
-    });
-
-    this.mainSegmentLoader_.on('progress', () => {
-      if (this.experimentalBufferBasedABR) {
+    if (!this.experimentalBufferBasedABR) {
+      this.mainSegmentLoader_.on('bandwidthupdate', () => {
         const nextPlaylist = this.selectPlaylist();
 
         if (this.shouldSwitchToMedia_(nextPlaylist)) {
           this.masterPlaylistLoader_.media(nextPlaylist);
         }
-      }
 
-      this.trigger('progress');
-    });
+        this.tech_.trigger('bandwidthupdate');
+      });
+
+      this.mainSegmentLoader_.on('progress', () => {
+        if (this.experimentalBufferBasedABR) {
```

Comment:
I think we can remove this if statement and everything inside of it now.

Reply:
yeah, this is unnecessary now.